README.md

```diff
@@ -146,6 +146,50 @@ locations.forEach((location) => {
 ```
 **[⬆ back to top](#table-of-contents)**
 
+### Avoid Mental Mapping of iterator states
+Nested high-performance loops can get confusing if you use incremental
+letters for iterator variables. Mental mapping overhead can be reduced
+when using the leading prefix letter (`d` for dataset) as the iterator
+and the same name + l (`dl` for dataset) as the length or loop's
```

Comment:
Can we remove the `dl` for dataset length. It does kind of go slightly against the don't over-optimize section.